includes/core/class-wc-payments-order.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Class file for WCPay\Core\WC_Payments_Order
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core;
+
+defined( 'ABSPATH' ) || exit;
+
+use \WC_Order;
+use WC_Payments_Utils;
+
+/**
+ * WC_Payments_Order
+ */
+class WC_Payments_Order {
+
+	/**
+	 * The WC_Order corresponding to this WC_Payments_Order.
+	 *
+	 * @var WC_Order
+	 */
+	private $order;
+
+	/**
+	 * Constructor when order_id is provided.
+	 *
+	 * @param WC_Order|int $order The order Id.
+	 */
+	public function __construct( $order ) {
+		$this->order = wc_get_order( $order );
+	}
+
+	/**
+	 * Get the payment metadata for key _intent_id.
+	 *
+	 * @return string
+	 */
+	public function get_payment_intent_id() : string {
+		return $this->order->get_meta( '_intent_id', true );
+	}
+
+
+	/**
+	 * Set the payment metadata for key _intent_id.
+	 *
+	 * @param  string $intent_id
+	 */
+	public function set_payment_intent_id( $intent_id ) {
+		$this->order->update_meta_data( '_intent_id', $intent_id );
+		 do_action( 'wcpay_payment_intent_id_updated' );
```

Comment:
Let's [document actions](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-hooks-actions-and-filters) to provide description.

Also, maybe including `order` in the action would make more sense, otherwise it looks like the payment intent is updated. So that would make it `wcpay_order_payment_intent_id_updated`. Combined with the method name change, it would turn to `wcpay_order_intent_id_updated`.

Reply:
Fixed